src/interactivity/win32/screenInfoUiaProvider.hpp

```diff
@@ -42,27 +44,30 @@ namespace Microsoft::Console::Interactivity::Win32
         void ChangeViewport(const SMALL_RECT NewWindow);
 
     protected:
-        std::deque<Microsoft::Console::Types::UiaTextRangeBase*> GetSelectionRanges(_In_ IRawElementProviderSimple* pProvider) override;
+        HRESULT GetSelectionRanges(_In_ IRawElementProviderSimple* pProvider, _Out_ std::deque<WRL::ComPtr<Microsoft::Console::Types::UiaTextRangeBase>>& selectionRanges) override;
 
         // degenerate range
-        Microsoft::Console::Types::UiaTextRangeBase* CreateTextRange(_In_ IRawElementProviderSimple* const pProvider) override;
+        HRESULT CreateTextRange(_In_ IRawElementProviderSimple* const pProvider, _COM_Outptr_result_maybenull_ Microsoft::Console::Types::UiaTextRangeBase** ppUtr) override;
 
         // degenerate range at cursor position
-        Microsoft::Console::Types::UiaTextRangeBase* CreateTextRange(_In_ IRawElementProviderSimple* const pProvider,
-                                                                     const Cursor& cursor) override;
+        HRESULT CreateTextRange(_In_ IRawElementProviderSimple* const pProvider,

```

Comment:
With all these methods returning `HRESULT` now, _can_ they be `noexcept`? (if they can, I'm not saying do that now, just file follow-up CodeHealth work)